### PR TITLE
Refactor - CreditCard purchase

### PR DESCRIPTION
**Pull request by tszulc**
- Replaces `expiration_date` with `expiryMonth` & `expiryYear` from CreditCard
- Adds required `origin` field
- Hard-codes some other _required_ fields (eg. `signature`, `pin`, `cvvStatus`)
- Determines the payment method based on the supported credit card brands
- Adds validation around `api_key`, `merchant_id`, `origin`, `amount`, `order`, and `description`
- Adds the response status code and message to the Response class
- Adds `isPending`, `message`, `code`, `transactionReference`, and `transactionId` to the Response class

### CODE REVIEW

File: src/Message/Response.php [hunk 1]
```diff
@@ -9,8 +9,145 @@
...
+    /**
+     * Is the response successful?
+     *
+     * @return boolean
+     */
     public function isSuccessful()
     {
-        // TODO: Implement isSuccessful() method.
+        if (isset($this->data['response']['data'][0]['status'])) {
+            return $this->data['response']['data'][0]['status'] != static::FAILED_STATUS;
```
Comment (jorbascrumps) [inline src/Message/Response.php:62]: Use strict equality for comparisons.

File: src/Message/AbstractRequest.php [hunk 2]
```diff
@@ -2,11 +2,86 @@
...
+     * Originated in a Payframe.
+     *
+     * @var int
+     */
+    const PAYFRAME_ORIGIN = 8;
+
+    /**
+     * Allowable values for the origin of the transaction.
+     *
+     * @link https://test-portal.payrix.com/docs/api#txnsPost
```
Comment (jorbascrumps) [inline src/Message/AbstractRequest.php:71]:
```suggestion
     * @link https://portal.payrix.com/docs/api#txnsPost
```

File: src/Message/PurchaseRequest.php [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+            'pin' => 0,
+            'origin' => $this->getOrigin(),
+            'currency' => $this->getCurrency(),
+        );
+
+        /**
+         * The total amount of this Transaction.
+         * This field is specified as an integer in cents.
+         *
+         * @link https://test-portal.payrix.com/docs/api#txnsPost
```
Comment (jorbascrumps) [inline src/Message/PurchaseRequest.php:60]:
```suggestion
         * @link https://portal.payrix.com/docs/api#txnsPost
```

**Review by jorbascrumps — CHANGES_REQUESTED** [review]

File: src/Message/Response.php (hunk 1 above)
Comment (tszulc) [reply src/Message/Response.php:62]:
```suggestion
            return $this->data['response']['data'][0]['status'] !== static::FAILED_STATUS;
```

**Review by jorbascrumps — APPROVED** [review]

**Pull request merged**